File: src/network/Encryption.ts
import CryptoJS from "crypto-js";

const KEY = CryptoJS.enc.Utf8.parse("mysecretkey12345");
const IV = CryptoJS.enc.Utf8.parse("0123456789abcdef");

export function encrypt(plaintext: string): string {
  const encrypted = CryptoJS.AES.encrypt(plaintext, KEY, {
    iv: IV,
    padding: CryptoJS.pad.Pkcs7,
    mode: CryptoJS.mode.CBC,
  });
  return encrypted.toString(); // Base64
}

export function decrypt(ciphertext: string): string {
  const decrypted = CryptoJS.AES.decrypt(ciphertext, KEY, {
    iv: IV,
    padding: CryptoJS.pad.Pkcs7,
    mode: CryptoJS.mode.CBC,
  });
  return decrypted.toString(CryptoJS.enc.Utf8);
}

File: src/python/encryption.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import base64

# Use a fixed 16-byte key (must be 16 bytes exactly)
KEY = b'mysecretkey12345'  # 16 bytes key
IV = b'0123456789abcdef'  # 16 bytes fixed IV

def encrypt(plaintext: str) -> str:
    cipher = AES.new(KEY, AES.MODE_CBC, IV)
    ct_bytes = cipher.encrypt(pad(plaintext.encode(), AES.block_size))
    return base64.b64encode(ct_bytes).decode()

def decrypt(ciphertext: str) -> str:
    ct = base64.b64decode(ciphertext)
    cipher = AES.new(KEY, AES.MODE_CBC, IV)
    pt = unpad(cipher.decrypt(ct), AES.block_size)
    return pt.decode()
